fmriprep/interfaces/multiecho.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+T2* map generation
+~~~~~~~~~~~~~~~~~~~~~~
+
+Using multi-echo EPI data, generates a T2*-map
+for use in T2*-driven EPI->T1 coregistration
+"""
+import os
+import numpy as np
+import nibabel as nb
+
+from niworkflows.nipype import logging
+from niworkflows.nipype.utils.filemanip import split_filename
+from niworkflows.nipype.interfaces.base import (
+    traits, TraitedSpec, File, InputMultiPath, SimpleInterface,
+    BaseInterfaceInputSpec)
+
+LOGGER = logging.getLogger('interface')
+
+
+class T2SMapInputSpec(BaseInterfaceInputSpec):
+    in_files = InputMultiPath(File(exists=True), mandatory=True,
+                              desc='multi-echo BOLD EPIs')
+    tes = traits.List(traits.Float, mandatory=True, desc='echo times')
+    compress = traits.Bool(True, usedefault=True, desc='use gzip compression on .nii output')
+
+
+class T2SMapOutputSpec(TraitedSpec):
+    output_image = File(exists=True, desc='T2* map')
+
+
+class T2SMap(SimpleInterface):
+    input_spec = T2SMapInputSpec
+    output_spec = T2SMapOutputSpec
+
+    def _run_interface(self, runtime):
+        ext = '.nii.gz' if self.inputs.compress else '.nii'
+        last_emask, two_emask = echo_sampling_mask(self.in_files)
+        t2s_map = define_t2s_map(self.in_files, self.tes,
+                                 last_emask, two_emask)
+        _, fname, _ = split_filename(self.inputs.in_files[0])
```

Comment:
I had not used `fname_presuffix` as that was capturing the 'echo-' key, which I wanted to drop since the T2* map pulls information across echos. If there's a way to have `fname_presuffix` do this, though, please let me know and I'll update appropriately!